### DISCUSSION TITLE
Iterate over key and value fields of json objects.

### DISCUSSION THREAD

**Discussion by GJanos** [stephenberry/glaze#352] [Ideas]
I checked and have not found a way to iterate over the key and value fields of json objects implemented inside glz yet.

One good use case would be, when you have a string containing a serialized json, (contents of the json are not known), and  would like to convert it to a glz::obj or glz::json_t (depending on the use case) instance. Then you would have to make operations (save / send / modify) on the individual fields. 
Something simmilar to nlohmann's approach could be cool.

```c+
        nlohmann::json json_obj = nlohmann::json::parse(json_str);

        for (nlohmann::json::iterator it = json_obj.begin(); it != json_obj.end(); ++it) {
            map_storing_json_fields[it.key()] = it.value().dump();
        }
``` 
But for this first glz would need a way to directly serialoze strings to glz::obj, just like how it currently does with glz::json_t.

Thank you for your time! If this feature is already supported and I was careless, please let me know.

**Comment by stephenberry** [comment]
@GJanos 
Both `glz::json_t` and `glz::obj` can be iterated in this way to build a map like your example. The `glz::obj` approach is a bit complex and should probably be cleaned up in the future.

I've written these in the form of unit tests:

For `glz::json_t`:
```c++
std::map<std::string, glz::raw_json> map;
glz::json_t obj{{ "arr", { 1, 2, 3 }}, { "hello", "world" }};
auto& o = obj.get<glz::json_t::object_t>();
for (auto&[key, value] : o) {
   map[key] = glz::write_json(value);
}

auto s = glz::write_json(map);
expect(s == R"({"arr":[1,2,3],"hello":"world"})") << s;
```

For `glz::obj`:
```c++
std::map<std::string, glz::raw_json> map;
glz::obj obj{ "arr", glz::arr{ 1, 2, 3 }, "hello", "world" };
using T = std::decay_t<decltype(obj.value)>;
glz::for_each<std::tuple_size_v<T>>([&](auto I) {
   if constexpr (I % 2 == 0) {
      map[std::string(glz::tuplet::get<I>(obj.value))] = glz::write_json(glz::tuplet::get<I + 1>(obj.value));
   }
});

auto s = glz::write_json(map);
expect(s == R"({"arr":[1,2,3],"hello":"world"})") << s;
```

**Reply by GJanos** [reply]
Thank you very much for letting me know. Indeed the glz::obj example would be better if simplified in the future. 
Why is the if constexpr (I % 2 == 0) { ...} part needed?

Are you planning on adding the feature of directly reading string into glz::obj?

**Reply by stephenberry** [reply]
Yes, I do plan to add parsing directly into `glz::obj`, just not sure when I'll get to it.

The `if constexpr (I % 2 == 0) { ...}` is needed because `glz::obj` is essentially a tuple, where every two items first refers to the key and then refers to the value. This is checking that we are accessing each pair of items instead of handling the key and value separately. even number value of I refer to the start of the pair. Odd values of I would incorrectly grab the value as the key.
